contracts/consensus/authority/Authority.sol

```diff
@@ -17,14 +17,21 @@ contract Authority is AbstractConsensus, Ownable {
 
     /// @notice Submit a claim.
     /// @param appContract The application contract address
+    /// @param lastProcessedBlockNumber The number of the last processed block
     /// @param claim The output Merkle root hash
     /// @dev Fires a `ClaimSubmission` event and a `ClaimAcceptance` event.
     /// @dev Can only be called by the owner.
     function submitClaim(
         address appContract,
+        uint256 lastProcessedBlockNumber,
         bytes32 claim
     ) external onlyOwner {
-        emit ClaimSubmission(msg.sender, appContract, claim);
-        _acceptClaim(appContract, claim);
+        emit ClaimSubmission(
+            msg.sender,
+            appContract,
+            lastProcessedBlockNumber,
```

Comment:
I assume that we have the same philosophy here that validator node can do whatever it wants. So `lastProcessedBlockNumber` may not be incremental.
Means the `ClaimSubmission` and `ClaimAcceptance` events may be emitted with a smaller or same `lastProcessedBlockNumber` compared with previous events, with same/different claims.

Reply:
Yes, here the `lastProcessedBlockNumber` parameter is purely for debugging purposes, and not checked against anything on-chain.